### PR TITLE
Migration from MSEng to DevDiv - release/1.1.0

### PR DESCRIPTION
**Pull request by ravimeda**
These changes migrate build of core-setup release 1.**1**.0 from the current MSEng to DevDiv.  Changes include:

1. Build definitions checked in under buildpipeline folder
2. Signing project is in build_projects/dotnet-host-build/sign.proj
3. Updated Agent Pool to DotNetCore-Test.
4. Fixed a yum package conflict.

@gkhanna79 please provide feedback.

### CODE REVIEW

**Comment by gkhanna79** [comment]
@dagood @wtgodbe @ramarag PTAL.

CC @leecow @eerhardt 

@ravimeda In general seems fine to me. Like the 1.0.0 PR, please make similar changes as suggested there and look into the Ubuntu x64 failure. also, we should hold off merging this until the branch reopens.

**Comment by ravimeda** [comment]
Thanks @gkhanna79.  I think the failures in Ubuntu16.04 x64 and OpenSUSE42.1 x64 are due to the missing _allowOverride_ as in 
`  "variables": {
    "BuildConfiguration": {
      "value": "Release",
      "allowOverride": true
    },`

**Comment by dagood** [comment]
The checked-in build defs shouldn't interact with CI at all. Also, CI errors are like:

```
00:00:03.897 /dev/stdin: line 2: syntax error near unexpected token `newline'
00:00:03.897 /dev/stdin: line 2: `<?xml version="1.0" encoding="utf-8"?>'
```

The only xml in this PR is `sign.proj`, but I don't know for sure if it's connected.

**Comment by ravimeda** [comment]
https://ci.dot.net/job/dotnet_core-setup/job/release_1.1.0/job/debug_opensuse42.1_x64_prtest/190/
https://ci.dot.net/job/dotnet_core-setup/job/release_1.1.0/job/release_ubuntu16.04_x64_prtest/193/
@dotnet-bot test this please.

File: buildpipeline/Core-Setup-RHEL7-x64.json [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+      "displayName": "Run build in Docker container",
+      "timeoutInMinutes": 0,
+      "task": {
+        "id": "6c731c3c-3c68-459a-a5c9-bde6e6595b5b",
+        "versionSpec": "2.*",
+        "definitionType": "task"
+      },
+      "inputs": {
+        "scriptPath": "scripts/dockerrun-as-current-user.sh",
+        "args": "-t --rm --sig-proxy=true --name $(DockerContainerName) -v $(Build.SourcesDirectory):/opt/code -w /opt/code -e CONNECTION_STRING -e PUBLISH_TO_AZURE_BLOB -e NUGET_FEED_URL -e NUGET_API_KEY -e GITHUB_PASSWORD chcosta/dotnetcore:rhel7_prereqs /bin/bash -c \"HOME=/opt/code; git clean -X -d -f; ./build.sh $(BuildArguments)\"",
```
Comment (eerhardt) [inline buildpipeline/Core-Setup-RHEL7-x64.json]: This should use the `microsoft/dotnet-buildtools-prereqs:rhel7_prereqs_2` docker image, which is the latest and has a "locale" issue fixed on it.

See https://github.com/dotnet/sdk/pull/935#issuecomment-285381546 for discussion.

I'm making the same change in `master` #1682.  This should be applied to the 1.0.0 change as well.

Comment (ravimeda) [inline buildpipeline/Core-Setup-RHEL7-x64.json]: Thanks @eerhardt

**Comment by ravimeda** [comment]
Commit `60051df` brings the new Docker image for RHEL to address a locale issue.  Refer to #1682

**Comment by ravimeda** [comment]
@MichaelSimons please could you review `60051df`

**Comment by MichaelSimons** [comment]
@ravimeda `60051df` LGTM

**Comment by gkhanna79** [comment]
My concern with moving to the new RH image is that it brings a new OS compared to what was used in the image used for previous servicing builds. As a result, the toolset can change as well.

Unless there is a compelling reason, I think we should keep the RHEL image the same as before since we want minimal (to zero) product/toolset changes as part of this transition. I would suggest undoing the commit.

**Comment by MichaelSimons** [comment]
@gkhanna79 - I completely understand your viewpoint.  I do however want to inform everyone with exactly what is different in the new `microsoft/dotnet-buildtools-prereqs:rhel7_prereqs_2` image that @eerhardt is suggesting we upgrade to.

If you run `docker inspect` you will see all of the layers are they same except a new one was added in `microsoft/dotnet-buildtools-prereqs:rhel7_prereqs_2`

```
docker inspect microsoft/dotnet-buildtools-prereqs:rhel7_prereqs
...
        "RootFS": {
            "Type": "layers",
            "Layers": [
                "sha256:13cde7f2a4831fb28e9b070a0b3858cbdbc71fb0f24937e3173b92918eb282ab",
                "sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef",
                "sha256:db56816e2d9709dc26218fa27eed4fe534e79dee20e29e3859e07a810d62a210",
                "sha256:d1c85d4a5755a0bf40486427db34988ee813391fa5c32acccbfe9a0a192bffb9",
                "sha256:4d7bf9f484aaaf7b2bf87281710e197e224c4a531c487e85302ce99fc853d429"
            ]
        }
```

```
docker inspect microsoft/dotnet-buildtools-prereqs:rhel7_prereqs_2
...
       "RootFS": {
           "Type": "layers",
           "Layers": [
               "sha256:13cde7f2a4831fb28e9b070a0b3858cbdbc71fb0f24937e3173b92918eb282ab",
               "sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef",
               "sha256:db56816e2d9709dc26218fa27eed4fe534e79dee20e29e3859e07a810d62a210",
               "sha256:d1c85d4a5755a0bf40486427db34988ee813391fa5c32acccbfe9a0a192bffb9",
               "sha256:4d7bf9f484aaaf7b2bf87281710e197e224c4a531c487e85302ce99fc853d429",
               **"sha256:4c1369084a7d3554b0f9b89445ca5f985131e2be2ae7c48c3500e3a140a29928"**
           ]
       }
```

Running `docker history microsoft/dotnet-buildtools-prereqs:rhel7_prereqs_2` will show you what that new layer is - `/bin/sh -c localedef -c -i en_US -f UTF-8 en_US.UTF-8`.  The OS and all of the packages are exactly the same.

**Comment by gkhanna79** [comment]
Thanks a lot for detailing the above @MichaelSimons - this affirms me our servicing builds have the right toolset. Given the data above, I am good with the merge of the two PRs.

**Pull request merged**